app/grandchallenge/algorithms/tasks.py

```diff
@@ -163,105 +159,137 @@ def create_algorithm_jobs(
+                    requires_memory_gb=requires_memory_gb,
+                    extra_viewer_groups=extra_viewer_groups,
+                    extra_logs_viewer_groups=extra_logs_viewer_groups,
+                    input_civ_set=ai.values.all(),
+                )
+                on_commit(job.execute)
 
-            jobs.append(job)
+                jobs.append(job)
 
     return jobs
 
 
-def filter_civs_for_algorithm(*, civ_sets, algorithm_image, algorithm_model):
+def filter_archive_items_for_algorithm(
+    *, archive_items, algorithm_image, algorithm_model=None
+):
     """
-    Removes sets of civs that are invalid for new jobs
+    Removes archive items that are invalid for new jobs.
+    The archive items need to contain values for all inputs of one of the algorithm's interfaces.
 
     Parameters
     ----------
-    civ_sets
-        Iterable of sets of ComponentInterfaceValues that are candidate for
-        new Jobs
+    archive_items
+        Archive items whose values are candidates for new jobs' inputs
     algorithm_image
         The algorithm image to use for new job
     algorithm_model
         The algorithm model to use for the new job or None
 
     Returns
     -------
-    Filtered set of ComponentInterfaceValues
+    Dictionary of valid ArchiveItems for new jobs, grouped by AlgorithmInterface
     """
     from grandchallenge.algorithms.models import Job
+    from grandchallenge.evaluation.models import (
+        get_archive_items_for_interfaces,
+        get_valid_jobs_for_interfaces_and_archive_items,
+    )
 
-    input_interfaces = {*algorithm_image.algorithm.inputs.all()}
+    if not archive_items:
+        return {}
 
-    existing_jobs = {
-        frozenset(j.inputs.all())
-        for j in Job.objects.filter(
-            algorithm_image=algorithm_image, algorithm_model=algorithm_model
+    archives = {ai.archive for ai in archive_items}
+    if len(archives) != 1:
+        raise RuntimeError(
+            "The archive items should all belong to the same archive."
         )
-        .annotate(
-            inputs_match_count=Count(
-                "inputs",
-                filter=Q(
-                    inputs__in={civ for civ_set in civ_sets for civ in civ_set}
-                ),
-            )
-        )
-        .filter(inputs_match_count=len(input_interfaces), creator=None)
-        .prefetch_related("inputs")
-    }
-
-    valid_job_inputs = []
-
-    for civ_set in civ_sets:
-        # Check interfaces are complete
-        civ_interfaces = {civ.interface for civ in civ_set}
-        if input_interfaces.issubset(civ_interfaces):
-            # If the algorithm works with a subset of the interfaces
-            # present in the set then only feed these through to the algorithm
-            valid_input = {
-                civ for civ in civ_set if civ.interface in input_interfaces
-            }
-        else:
-            continue
+    (archive,) = archives
```

Comment:
For my understanding, what does `(archive,) = archives` do?

Reply:
`archives` is a set that contains 1 archive, and this extracts that archive so I can use it in the filter. `*archives` is not supported in queries, so this seemed like the best way to extract the archive. But it is anyway not needed anymore given that I'm now filtering by archive_items directly.